src/Facebook/FileUpload/FacebookVideoUploader.php

```diff
@@ -0,0 +1,255 @@
+     * Upload video file by chunks
+     *
+     * @param string       $filePath
+     * @param int          $targetId
+     * @param string       $accessToken
+     * @param int          $maxTransferTries
+     * @return array
+     */
+    public function upload($filePath, $targetId, $accessToken, $maxTransferTries = 3)
+    {
+        //validate params
+        $res = [];
+        if (empty($this->facebook) || empty($filePath) || empty($targetId) || empty($accessToken)) {
+            $res['errorMsg'] = 'Missing required params: ' .
+                (empty($this->facebook) ? 'facebook ' : '') .
+                (empty($accessToken) ? 'accessToken ' : '') .
+                (empty($filePath) ? 'filePath ' : '') .
+                (empty($targetId) ? 'targetId ' : '');
+
+            $res['success'] = false;
+            return $res;
+        }
+
+        //validate files
+        if (!file_exists($filePath)) {
+            $res['error'] = [
+                'message' => 'File not exists',
+            ];
+            $res['success'] = false;
+            return $res;
+        }
+
+        //set up context
+        $this->context = new VideoUploadContext();
+        $this->context->filePath = $filePath;
+        $this->context->fileSize = filesize($filePath);
+        $this->context->targetId = $targetId;
+        $this->context->accessToken = $accessToken;
+
+        //setup max retries
+        $this->maxTransferTries = $maxTransferTries;
+
+        try {
+            //Start - Initialize an upload session
+            $this->uploadStart();
+
+            //Transfer - Upload video chunks
+            $this->uploadTransfer();
+
+            //Finish - Post the video
+            $this->uploadFinish();
+            $res['success'] = true;
+            $res['video_id'] = $this->context->videoId;
+        } catch (FacebookResponseException $e) {
+            $res['success'] = false;
+            $res['error'] = isset($e->getResponseData()['error']) ? $e->getResponseData()['error'] : array();
+        } catch (FacebookSDKException $e) {
+            $res['error'] = [
+                'message' => $e->getMessage(),
+            ];
+            $res['success'] = false;
+        }
+
+        return $res;
+    }
+
+    /**
+     * Upload - phase start
+     */
+    public function uploadStart()
+    {
+        $startReqParams = array(
+            'upload_phase' => 'start',
+            'file_size' => $this->context->fileSize,
+        );
+
+        $res = $this->facebook->post('/'.$this->context->targetId.'/videos',
+            $startReqParams,
+            $this->context->accessToken
+        )->getDecodedBody();
+
+        $this->context->uploadSessionId = $res['upload_session_id'];
+        $this->context->startOffset = $res['start_offset'];
+        $this->context->endOffset = $res['end_offset'];
+        $this->context->videoId = $res['video_id'];
+    }
+
+    /**
+     * Upload - phase transfer
+     */
+    public function uploadTransfer()
```

Comment:
@YuanhuiChen ,please set this function as private. Since this function is working on one operation, it might be better to keep things together for readability.